cmp/filter/FilterChooserModel.js

```diff
@@ -0,0 +1,606 @@
+    parseRawFieldSpecs(rawSpecs) {
+        const {store} = this,
+            ret = [];
+
+        // If no specs provided, include all Store Fields.
+        if (isEmpty(rawSpecs)) rawSpecs = store.fieldNames;
+
+        rawSpecs.forEach(spec => {
+            if (isString(spec)) spec = {field: spec};
+            const storeField = store.getField(spec.field);
+
+            if (!storeField) {
+                console.warn(`Field '${spec.field}' not found in linked Store - will be ignored.`);
+                return;
+            }
+
+            ret.push({
+                ...spec,
+                field: storeField
+            });
+        });
+
+        return ret;
+    }
+
+    getFieldSpec(fieldName) {
+        return this.fieldSpecs.find(it => it.field.name == fieldName);
+    }
+}
+
+/**
+ * @typedef {Object} FilterChooserFieldSpecConfig - developer-facing config API to customize
+ *      filtering behavior at the Field level.
+ * @property {string} field - name of Store Field to enable for filtering - must be resolvable to a
+ *      known Field within the associated Store.
+ * @property {string} [displayName] - optional override for `Field.displayName` for use within
+ *      filtering component controls.
+ * @property {string[]} [operators] - operators available for filtering. Optional, will default to
+ *      a supported set based on the type of the provided Field.
+ * @property {boolean} [suggestValues] - true to provide auto-complete options with data
+ *      values sourced either automatically from Store data or as provided directly via the
+ *      `values` config below. Default `true` when supported based on the type of the provided
+ *      Field and operators. Set to `false` to disable extraction/suggestion of values from Store.
+ * @property {[]} [values] - explicit list of available values to autocomplete for this Field.
+ *      Optional, will otherwise be extracted and updated from available Store data if applicable.
+ * @property {FilterOptionValueRendererCb} [valueRenderer] - function to produce a suitably
+ *      formatted string for display to the user for any given field value.
+ * @property {FilterOptionValueParserCb} [valueParser] - function to parse user's input from a
+ *      filter chooser control into a typed data value suitable for use in filtering comparisons.
+ * @property {*} [exampleValue] - sample / representative value used by components to aid usability.
+ */
+
+/**
+ * @typedef {Object} RawFilterChooserFieldSpec - partially processed spec for internal use by this
+ *     class. Identical to `FilterChooserFieldSpecConfig` but with resolved `Field` instance.
+ * @property {Field} field
+ * @property {string} displayName
+ * @property {string[]} operators
+ * @property {boolean} suggestValues
+ * @property {[]} values
+ * @property {FilterOptionValueRendererCb} [valueRenderer]
+ * @property {FilterOptionValueParserCb} [valueParser]
+ * @property {*} [exampleValue]
+ */
```

Comment:
I fully admit that maintaining/typing this intermediate object is inelegant.  This could be cleaned up if we decided to create `FilterChooserFieldSpec`s once and then let them internally update in response to store changes. We keep these processed specs around in this model so we can pre-process once and then re-create as needed.

And this is also a case where the typing is not for public documentation but really helped me stay sane and keep track of what these objects have - especially with 'field' which can reference both field names and Field instances throughout the codebase.